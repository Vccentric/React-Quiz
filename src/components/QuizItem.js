import React from "react";
import PropTypes from "prop-types";

// quiz item component
const QuizItem = ({ quiz, setView }) => {
  // setup variables
  const name = quiz.name || "";
  const total = quiz.questions?.length || 0;

  // setup functions
  const onClick = () =>
    setView({
      showQuizList: false,
      showQuiz: quiz,
    });

  return (
    <div className="quiz" onClick={onClick}>
      <div className="name">{name}</div>
      <div className="total">{total} Questions</div>
    </div>
  );
};

QuizItem.propTypes = {
  quiz: PropTypes.object,
  setView: PropTypes.func,
};

export default QuizItem;
