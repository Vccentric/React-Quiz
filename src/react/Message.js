import React from 'react';
import { hot } from 'react-hot-loader';

// message component
function Message(props) {
    const { isAnswered, isCorrect } = props;
    return (
        <div
            id="message"
            style={isAnswered ? { display: 'block' } : { display: 'none' }}
            className={isCorrect ? 'correct' : 'wrong'}
        >
            {isCorrect ? 'Correct Answer!' : 'Wrong Answer!'}
        </div>
    );
}

export default hot(module)(Message);
