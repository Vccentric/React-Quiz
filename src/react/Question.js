import React from 'react';
import { hot } from 'react-hot-loader';

// question component
function Question(props) {
    const { question } = props;
    return <p id="question">{question}</p>;
}

export default hot(module)(Question);
