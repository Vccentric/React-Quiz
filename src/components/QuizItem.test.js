import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import React from "react";
import QuizItem from "./QuizItem";

// setup variables
const quiz = { name: "Quiz Name", questions: [1] };
const setQuiz = jest.fn();

test("renders correctly - snapshot", () => {
  const tree = renderer
    .create(<QuizItem quiz={quiz} setQuiz={setQuiz} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders quiz item with empty quiz object", () => {
  const { container } = render(<QuizItem quiz={{}} setQuiz={setQuiz} />);
  const quizNameText = container.getElementsByClassName("name");
  const totalQuestionText = container.getElementsByClassName("total");
  expect(quizNameText?.[0]?.textContent).toBe("");
  expect(totalQuestionText?.[0]?.textContent).toBe("0 Questions");
});

test("renders quiz item with defined quiz object", () => {
  render(<QuizItem quiz={quiz} setQuiz={setQuiz} />);
  const quizNameText = screen.getByText("Quiz Name");
  const totalQuestionText = screen.getByText("1 Questions");
  expect(quizNameText).toBeInTheDocument();
  expect(totalQuestionText).toBeInTheDocument();
});
