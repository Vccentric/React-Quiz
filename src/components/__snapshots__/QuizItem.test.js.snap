// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - snapshot 1`] = `
<div
  className="quiz"
  onClick={[Function]}
>
  <div
    className="name"
  >
    Quiz Name
  </div>
  <div
    className="total"
  >
    1
     Questions
  </div>
</div>
`;
